Modules/_asynciomodule.c

```diff
@@ -962,14 +962,18 @@ FutureObj_dealloc(PyObject *self)
 {
     FutureObj *fut = (FutureObj *)self;
 
+    PyObject_GC_UnTrack(self);
+
     if (Future_CheckExact(fut)) {
```

Comment:
k, I'm claiming ignorance since I don't know what these macros achieve :)  I'd personally feel safer if we had a test case, do you know how to create a testcase to cause this to crash?  Mine takes a day to reproduce in a production environment :(